components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +132,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.where(column, OperatorType.LIKE, "%");
```

Comment:
We had an offline investigation, and the reason was that the column type was VARCHAR2 (note the *2*). The main feature of that column is that null and empty strings are treated equally.